package/MDAnalysis/analysis/polymer.py

```diff
@@ -27,61 +27,158 @@
 from .base import AnalysisBase
 
 logger = logging.getLogger(__name__)
 
 
+@requires('bonds')
+def sort_backbone(backbone):
+    """Rearrange a linear AtomGroup into backbone order
+
+    Requires that the backbone has bond information,
+    and that only backbone atoms are provided (ie no side
+    chains or hydrogens).
+
+    Parameters
+    ----------
+    backbone : AtomGroup
+      the backbone atoms, not necessarily in order
+
+    Returns
+    -------
+    sorted_backbone : AtomGroup
+      backbone in order, so `sorted_backbone[i]` is bonded to
+      `sorted_backbone[i - 1]` and `sorted_backbone[i + 1]`
+
+
+    .. versionadded:: 0.20.0
+    """
+    if not len(backbone.fragments) == 1:
+        raise ValueError("Multiple fragments found in backbone.  "
+                         "backbone must be a single contiguous AtomGroup")
+    if max(len(at.bonded_atoms & backbone) for at in backbone) > 2:
+        # find which atom has too many bonds for easier debug
+        branches = [at for at in backbone
+                    if len(at.bonded_atoms & backbone) > 2]
+        raise ValueError(
+            "backbone is not linear.  "
+            "The following atoms have more than two bonds in backbone: "
+            "{}".format(','.join(str(a) for a in branches)))
+
+    caps = [atom for atom in backbone
+           if len(atom.bonded_atoms & backbone) == 1]
+
+     # choose one of the capping hydrogens to be the startpoint
+    sorted_backbone = AtomGroup([caps[0]])
+
+    # iterate until the sorted chain length matches the backbone size
+    while len(sorted_backbone) < len(backbone):
+        # current end of the chain
+        end_atom = sorted_backbone[-1]
+
+        # look at all bonded atoms which are also part of the backbone
+        # and subtract any that have already been added
+        next_atom = (end_atom.bonded_atoms & backbone) - sorted_backbone
+
+        # append this to the sorted backbone
+        sorted_backbone += next_atom
+
+    return sorted_backbone
+
+
 class PersistenceLength(AnalysisBase):
     r"""Calculate the persistence length for polymer chains
 
     The persistence length is the length at which two points on the polymer
-    chain become decorrelated.
+    chain become decorrelated.  This is determined by first measuring the
+    autocorrelation (:math:`C(n)`) of two bond vectors (:math:`\mathbf{a}`)
+    separated by :math:`n` bonds
 
-    Notes
-    -----
-    This analysis requires that the trajectory supports indexing.
+    .. math::
+
+       C(n) = \langle \cos\theta_{i, i+n} \rangle =
+               \langle \mathbf{a_i} \cdot \mathbf{a_{i+n}} \rangle
+
+    An exponential decay is then fitted to this, which yields the
+    persistence length
+
+    .. math::
+
+       C(n) \approx \exp\left( - \frac{n l_B}{l_P} \right)
+
+    where :math:`l_B` is the average bond length, and :math:`l_P` is the
+    persistence length.
 
+    Parameters
+    ----------
+    atomgroups : list
```

Comment:
shouldn't that be an `iterable`? Also works with tuples or object arrays i think.